packages/store/src/StoreClass.tsx

```diff
@@ -253,6 +255,63 @@ export class StoreClass<T extends MachinesObj> implements IStore<T> {
     const service = this.services[key];
     return waitFor(service, ...args);
   }
+
+  /**
+   * Wait for a specifice state to be reached.
+   * @param key - the key of the service
+   * @param done - the state to wait for complete (default: 'done')
+   * @param failure - the state to wait for failure (default: 'failed')
+   * @param failureMessage - the field of the context to use as the error message (default: 'error')
+   * @param timeout - the timeout in milliseconds (default: 5000)
+   * @returns the current state value
+   * @example
+   * await store.waitForState('counter');
+   * // or
+   * await store.waitForState('counter', 'done', 'failed', 'error');
+   */
+  public async waitForState<K extends keyof T, S extends StateItem<T, K>>(
+    ...[
+      key,
+      {
+        done = 'done',
+        failure = 'failed',
+        failureMessage = 'error',
+        timeout = 5000,
+      } = {},
+    ]: WaitForStateArgs<T, K>
+  ) {
+    try {
+      const service = this.services[key];
+      const doneState = done || 'done';
+      const failureState = failure || 'failed';
+      const failureMessageField = failureMessage || 'error';
```

Comment:
```suggestion
      const doneState = done';
      const failureState = failure;
      const failureMessageField = failureMessage;
```

Can we do this bc of the default assignments in the function definition?

Reply:
I think we should keep because default assignments in function definition will work unless you define something.... so for example: `done: ''` would bypass it